sdk/cosmosdb/cosmos/src/queryExecutionContext/pipelinedQueryExecutionContext.ts

```diff
@@ -50,10 +50,24 @@ export class PipelinedQueryExecutionContext implements ExecutionContext {
     const sortOrders = partitionedQueryExecutionInfo.queryInfo.orderBy;
     // TODO: Currently we don't get any field from backend to determine streaming queries
     if (this.nonStreamingOrderBy) {
+      if (!options.allowUnboundedNonStreamingQueries) {
+        this.checkQueryConstraints(partitionedQueryExecutionInfo.queryInfo);
+      }
+
       this.vectorSearchBufferSize = this.calculateVectorSearchBufferSize(
         partitionedQueryExecutionInfo.queryInfo,
         options,
       );
+      const maxBufferSize = options["vectorSearchBufferSize"]
+        ? options["vectorSearchBufferSize"]
+        : PipelinedQueryExecutionContext.DEFAULT_MAX_VECTOR_SEARCH_BUFFER_SIZE;
+
+      if (this.vectorSearchBufferSize > maxBufferSize) {
+        throw new ErrorResponse(
+          `Executing a vector search query with TOP or LIMIT larger than the vectorSearchBufferSize ${maxBufferSize} ` +
```

Comment:
also log ${vectorSearchBufferSize }

Reply:
that is already known to Cx in TOP or LIMIT value